src/test/java/com/cloudbees/jenkins/support/SupportTestUtils.java

```diff
@@ -16,23 +16,48 @@
  */
 public class SupportTestUtils {
 
-  /**
-   * Invoke a component, and return the component contents as a String.
-   */
-  public static String invokeComponentToString (final Component component) {
-    final ByteArrayOutputStream baos = new ByteArrayOutputStream();
-    component.addContents(
-            new Container() {
-              @Override
-              public void add(@CheckForNull Content content) {
-                try {
-                  content.writeTo(baos);
-                } catch (IOException e) {
-                  e.printStackTrace();
-                }
-              }
-            });
+    /**
+     * Invoke a component, and return all component contents as a String.
+     * Useful to unit test a specific component.
+     */
+    public static String invokeComponentToString(final Component component) {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        component.addContents(
+                new Container() {
+                    @Override
+                    public void add(@CheckForNull Content content) {
+                        try {
+                            content.writeTo(baos);
+                        } catch (IOException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                });
 
-    return baos.toString();
-  }
+        return baos.toString();
+    }
+
+    /**
+     * Invoke a component, and return a component content as a String.
+     * Useful to unit test a specific content.
+     */
+    public static String invokeContentToString(final Component component, final String contentName) {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        component.addContents(
+                new Container() {
+                    @Override
+                    public void add(@CheckForNull Content content) {
+                        if(!contentName.equals(content.getName())) {
```

Comment:
Why this check?

Reply:
I wanted to just grab a specific content rather than a full component. I reverted this.